src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -113,6 +126,46 @@ public void notifyChangedTestCases(ProgrammingExercise programmingExercise, Obje
         }
     }
 
+    public void addStudentIdToProjectName(Repository repo, ProgrammingExercise programmingExercise, Participation participation) {
+        // is Java programming language
+        if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.JAVA) {
+            // Eclipse .project file
+            File eclipseProjectFile = new File(repo.getLocalPath().toString(), ".project");
+            if (eclipseProjectFile.exists()) {
+
+                try {
+                    // 1- Build the doc from the XML file
+                    Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(new InputSource(eclipseProjectFile.getPath()));
+                    doc.setXmlStandalone(true);
+
+                    // 2- Find the node with xpath
+                    XPath xPath = XPathFactory.newInstance().newXPath();
+                    Node nameNode = (Node) xPath.compile("/projectDescription/name").evaluate(doc, XPathConstants.NODE);
+
+                    // 3- Set new value
+                    nameNode.setTextContent(programmingExercise.getTitle() + " " + participation.getStudent().getLogin());
```

Comment:
What happens if the original name was different than the programmingExercise title?

I think it would be better to use the existing name and simply append the student login

Reply:
https://github.com/ls1intum/ArTEMiS/pull/491/commits/15e0116fe83a0e10cd5eb04e274a6ae0c9646f7d#diff-f4c3e69e07b0f66f35b2d3a3fce450eeR146